src/ngx_http_lua_connection.c

```diff
@@ -0,0 +1,353 @@
+/*
+ * Copyright (C) 2014-2015 Daurnimator
+ */
+
+/* need to include DDEBUG so that ngx_http_lua_util.h works */
+#ifndef DDEBUG
+#define DDEBUG 0
+#endif
+#include "ddebug.h"
+
+#include "ngx_http_lua_connection.h"
+
+#include <math.h> /* HUGE_VAL */
+#include <poll.h> /* POLLIN, POLLOUT */
+
+#include <lua.h>
+#include <lauxlib.h>
+
+#include "ngx_core.h"
+#include "ngx_alloc.h" /* ngx_alloc, ngx_free */
+#include "ngx_http.h" /* ngx_http_request_t, ngx_http_run_posted_requests */
+#include "ngx_http_lua_common.h" /* ngx_http_lua_module, ngx_http_lua_co_ctx_t */
+
+#include "ngx_http_lua_util.h" /* ngx_http_lua_get_lua_vm, ngx_http_lua_run_thread, ngx_http_lua_finalize_request */
+#include "ngx_http_lua_contentby.h" /* ngx_http_lua_content_wev_handler */
+
+
+typedef struct {
+    ngx_http_request_t    *request;
+    ngx_http_lua_co_ctx_t *co_ctx;
+    ngx_connection_t      *conn;
+} ngx_http_lua_udata_t;
+
+
+static ngx_int_t
+ngx_http_lua_fd_resume_request(ngx_http_request_t *r)
```

Comment:
@daurnimator Ah, I prefer something shorter like `conn` since seeing long prefixes like `ngx_http_lua_conneciton_` all over the place would make me crazy ;)